Plugins/Sycl/src/Seeding/CreateSeedsForGroupSycl.cpp

```diff
@@ -212,133 +248,181 @@ void createSeedsForGroupSycl(
-
       // We store the indices of the BOTTOM/TOP space points of the edges of
       // the bottom-middle and top-middle bipartite duplet graphs. They index
       // the bottomSPs and topSPs vectors.
-      auto deviceIndBot = make_device_array<uint32_t>(edgesBottom, *q);
-      auto deviceIndTop = make_device_array<uint32_t>(edgesTop, *q);
 
       // We store the indices of the MIDDLE space points of the edges of the
       // bottom-middle and top-middle bipartite duplet graphs.
       // They index the middleSP vector.
-      auto deviceMidIndPerBot = make_device_array<uint32_t>(edgesBottom, *q);
-      auto deviceMidIndPerTop = make_device_array<uint32_t>(edgesTop, *q);
+      // indMidBotComp;
+      // indMidTopComp;
 
       // Partial sum arrays of deviceNumBot and deviceNum
-      auto deviceSumBot = make_device_array<uint32_t>(M + 1, *q);
-      auto deviceSumTop = make_device_array<uint32_t>(M + 1, *q);
-
       // Partial sum array of the combinations of compatible bottom and top
       // space points per middle space point.
-      auto deviceSumComb = make_device_array<uint32_t>(M + 1, *q);
-
       // Allocations for coordinate transformation.
-      auto deviceLinBot =
-          make_device_array<detail::DeviceLinEqCircle>(edgesBottom, *q);
-      auto deviceLinTop =
-          make_device_array<detail::DeviceLinEqCircle>(edgesTop, *q);
-
-      q->memcpy(deviceMidIndPerBot.get(), indMidBotComp.data(),
-                sizeof(uint32_t) * edgesBottom);
-      q->memcpy(deviceMidIndPerTop.get(), indMidTopComp.data(),
-                sizeof(uint32_t) * edgesTop);
-      q->memcpy(deviceSumBot.get(), sumBotCompUptoMid.data(),
-                sizeof(uint32_t) * (M + 1));
-      q->memcpy(deviceSumTop.get(), sumTopCompUptoMid.data(),
-                sizeof(uint32_t) * (M + 1));
-      q->memcpy(deviceSumComb.get(), sumBotTopCombined.data(),
-                sizeof(uint32_t) * (M + 1));
-      q->wait();
 
+
+      // Buffers for Flattening the jagged vectors
+      std::unique_ptr<vecmem::data::vector_buffer<uint32_t>> indBotDupletBuffer;
+      std::unique_ptr<vecmem::data::vector_buffer<uint32_t>> indTopDupletBuffer;
+      if (!device_resource){
+        indBotDupletBuffer = std::make_unique<vecmem::data::vector_buffer<uint32_t>>
+                                                              (edgesBottom, resource);
+        indTopDupletBuffer = std::make_unique<vecmem::data::vector_buffer<uint32_t>>
+                                                              (edgesTop, resource);
+      } else {
+        indBotDupletBuffer = std::make_unique<vecmem::data::vector_buffer<uint32_t>>
+                                                              (edgesBottom, *device_resource);
+        indTopDupletBuffer = std::make_unique<vecmem::data::vector_buffer<uint32_t>>
+                                                              (edgesTop, *device_resource);
+      }
```

Comment:
Similar to my earlier comment, this could be simplified a bit with `?` conditional operators.

Reply:
Done.